### PR TITLE
fix type handling for clocktime and step_size

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  other
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5480

### CODE REVIEW

**Review by hussain-jafari — APPROVED** [review]
approved assuming commented out code will be removed

File: src/vivarium/framework/event.py [hunk 1]
```diff
@@ -115,11 +115,28 @@ def emit(self, index: pd.Index[int], user_data: dict[str, Any] | None = None) ->
...
+            raise ValueError(
+                f"Clock ({type(clock)}) and step size ({type(step_size)}) are not compatible."
+            )
+
         e = Event(
             index,
             user_data,
-            self.manager.clock() + self.manager.step_size(),  # type: ignore[operator, arg-type]
-            self.manager.step_size(),
+            # self.manager.clock() + self.manager.step_size(),
```
Comment (rmudambi) [inline src/vivarium/framework/event.py:136]: Remove these commented lines

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,3 +1,7 @@
+**3.0.16 - 10/31/24**
```
Comment (hussain-jafari) [inline CHANGELOG.rst:1]: is this intended to be tomorrow?

**Review by rmudambi — APPROVED** [review]

File: CHANGELOG.rst (hunk 2 above)
Comment (stevebachmeier) [reply CHANGELOG.rst:1]: Yes just b/c I rarely get approvals this late in the day

**Review by albrja — APPROVED** [review]

File: src/vivarium/framework/event.py [hunk 3]
```diff
@@ -115,11 +115,26 @@ def emit(self, index: pd.Index[int], user_data: dict[str, Any] | None = None) ->
         """
         if not user_data:
             user_data = {}
+
+        clock = self.manager.clock()
+        step_size = self.manager.step_size()
+        event_time: ClockTime
+        if isinstance(clock, int) and isinstance(step_size, int):
+            event_time = clock + step_size
+        elif isinstance(clock, (pd.Timestamp, datetime)) and isinstance(
```
Comment (rmudambi) [inline src/vivarium/framework/event.py:124]: Apparently you can just do datetime and timedelta, since the pandas types inherit from them

**Review by albrja — APPROVED** [review]

**Pull request merged**